metricshub-snmpv3-extension/src/main/java/org/sentrysoftware/metricshub/extension/snmpv3/SnmpV3Configuration.java

```diff
@@ -0,0 +1,216 @@
+	private char[] password;
+	private int[] retryIntervals;
+
+	@Override
+	public String toString() {
+		String description = "SNMP V3";
+		if (username != null) {
+			description = description + " as " + username;
+		}
+		if (privacy != null && privacy != Privacy.NO_ENCRYPTION) {
+			description = description + " (" + privacy + "-encrypted)";
+		}
+		return description;
+	}
+
+	@Override
+	public void validateConfiguration(final String resourceKey) throws InvalidConfigurationException {
+		StringHelper.validateConfigurationAttribute(
+			community,
+			attr -> attr == null || attr.length == 0,
+			() ->
+				String.format(
+					"Resource %s - No community string configured for %s. This resource will not be monitored.",
+					resourceKey,
+					community
+				)
+		);
+
+		StringHelper.validateConfigurationAttribute(
+			port,
+			attr -> attr == null || attr < 1 || attr > 65535,
+			() ->
+				String.format(
+					"Resource %s - Invalid port configured: %s. Please verify the configured port value.",
+					resourceKey,
+					port
+				)
+		);
+
+		StringHelper.validateConfigurationAttribute(
+			timeout,
+			attr -> attr == null || attr < 0L,
+			() ->
+				String.format(
+					"Resource %s - Timeout value is invalid: %s. Please verify the configured timeout value.",
+					resourceKey,
+					timeout
+				)
+		);
+
+		StringHelper.validateConfigurationAttribute(
+			username,
+			attr -> attr == null || attr.isEmpty(),
+			() ->
+				String.format(
+					"Resource %s - No username configured for protocol %s." +
+					" This resource will not be monitored. Please verify the configured username.",
+					resourceKey,
+					username
+				)
+		);
+
+		StringHelper.validateConfigurationAttribute(
+			authType,
+			attr -> attr == null,
+			() ->
+				String.format(
+					"Resource %s - No username configured for protocol %s." +
+					" This resource will not be monitored. Please verify the configured authtype.",
+					resourceKey,
+					authType
+				)
+		);
+	}
+
+	/**
+	 * Enum of authentication types for SNMP v3.
+	 */
+	@AllArgsConstructor
+	public enum AuthType {
+		NO_AUTH("NO_AUTH"),
+		MD5("MD5"),
+		SHA("SHA");
+
+		@Getter
+		private final String stringAuthType;
+
+		/**
+		 * Interpret the specified label and returns corresponding value.
+		 *
+		 * @param authType String to be interpreted
+		 * @return Corresponding {@link AuthType} value
+		 */
+		public static AuthType interpretValueOf(@NonNull final String authType) {
+			final String lowerCaseAuthType = authType.toLowerCase();
+
+			for (AuthType type : AuthType.values()) {
+				if (type.getStringAuthType().equals(lowerCaseAuthType)) {
+					return type;
+				}
+			}
+
+			throw new IllegalArgumentException(INVALID_AUTH_TYPE_EXCEPTION_MESSAGE + authType);
+		}
+	}
+
+	/**
+	 * SNMP v3 Privacy (encryption type). Represents the encryption algorithm to be
+	 * used in SNMP v3 connections.
+	 */
+	@AllArgsConstructor
+	public enum Privacy {
+		NO_ENCRYPTION("NO_ENCRYPTION"),
```

Comment:
`stringAuthType` is not required since it shows the same string value as the enumeration. 
The interpretValueOf should perform the following interpretation:
```java
			if (lCasePrivacy.equals("none") || lCasePrivacy.equals("no")) {
				return NO_ENCRYPTION;
			}
```